### PR TITLE
Update single_source_dijkstra_path and all_pairs_dijkstra_path

### PR DESCRIPTION
**Pull request by MridulS**
@dschult

### CODE REVIEW

**Comment by dschult** [comment]
This looks like it doesn't conflict with #1851 but they both change the same file (weighted.py). I'm going to merge that request before these conflict.  You may have to do a rebase. Sorry about that.

File: networkx/algorithms/shortest_paths/weighted.py [hunk 1]
```diff
@@ -174,7 +174,7 @@ def single_source_dijkstra_path(G, source, cutoff=None, weight='weight'):
     """
     (length, path) = single_source_dijkstra(
         G, source, cutoff=cutoff, weight=weight)
-    return path
+    return iter(path.items())
```
Comment (jfinkels) [inline networkx/algorithms/shortest_paths/weighted.py]: Why `iter` necessary here? Isn't `path.items()` already iterable? That should be enough for the user to write code that uses the return value of this function, expecting an iterable.

**Comment by hagberg** [comment]
Is this still needed?

**Comment by dschult** [comment]
This is not still needed. Our turning the shortest_path functions into iterators needs more thought.

We should not create a dict holding the path only to iterate through the entries (and then often create another dict from the iterates). To make these routines iterators that report as they compute, we would need to either split out the length/pred/path algorithms (as they once were).  Each would then yield the appropriate output. Or we could keep the algorithm in one helper function (as now) but yield the appropriate results for possible all three of length/pred/path upon discovery of each node.  If ```paths is None``` then we would not yield paths; if ```pred is None``` we would not yield preds.

Or we can leave it as is: shortest_path_length routines iterate, shortest_path routines return dicts.

**Comment by hagberg** [comment]
Suggest leaving this as is unless a good reason to change is brought.

**Pull request closed without merging**